airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/stream_reader.py

```diff
@@ -197,7 +197,15 @@ def drives(self):
         """
         Retrieves and caches SharePoint drives, including the user's drive based on authentication type.
         """
-        drives = execute_query_with_retry(self.one_drive_client.drives.get())
+drives = []
+        sites = execute_query_with_retry(self.one_drive_client.sites.get_all_sites().execute_query())
```

Comment:
best if we can define sites when defining connector. tbh should be connector per site